opentelemetry-sdk/src/logs/log_processor.rs

```diff
@@ -410,37 +484,44 @@ impl BatchLogProcessor {
                         .unwrap_or(config.scheduled_delay);
 
                     match message_receiver.recv_timeout(remaining_time) {
-                        Ok(BatchMessage::ExportLog(log)) => {
-                            logs.push(log);
-                            if logs.len() == config.max_export_batch_size {
-                                otel_debug!(
-                                    name: "BatchLogProcessor.ExportingDueToBatchSize",
-                                );
-                                let _ = export_with_timeout_sync(
-                                    config.max_export_timeout,
-                                    &mut exporter,
-                                    &mut logs,
-                                    &mut last_export_time,
-                                );
-                            }
+                        Ok(BatchMessage::ExportLog(export_log_message_sent)) => {
+                            otel_debug!(
+                                name: "BatchLogProcessor.ExportingDueToBatchSize",
+                            );
+
+                            let _ = get_logs_and_export(
+                                &logs_receiver,
+                                &exporter,
+                                &mut logs,
+                                &mut last_export_time,
+                                &current_batch_size,
+                                &config,
+                            );
+
+                            // Reset the export log message sent flag now it has has been processed.
+                            export_log_message_sent.store(false, Ordering::Relaxed);
                         }
                         Ok(BatchMessage::ForceFlush(sender)) => {
                             otel_debug!(name: "BatchLogProcessor.ExportingDueToForceFlush");
-                            let result = export_with_timeout_sync(
-                                config.max_export_timeout,
-                                &mut exporter,
+                            let result = get_logs_and_export(
```

Comment:
this will not be sufficient as it'll only export upto max_export_batch_size logs. We need to export everything received upto flush invocation.

Reply:
> We need to export everything received upto flush invocation.

That should be taken care of by the signal sent by the logging thread whenever we have `max_export_batch_size` number of logs. That means that the background thread would be working on the signal sent by the logger thread before working on ForceFlush/Shutdown thread.

Reply:
Had an offline discussion about this with @cijothomas, and we agreed that there is a possibility of logs just residing in the channel without being exported. This should be fixable. I would address this in the next PR.

Reply:
resolving for follow up.